ui/src/app/kvm/components/KVMActionFormWrapper/ComposeForm/StorageTable/PoolSelect/PoolSelect.tsx

```diff
@@ -0,0 +1,179 @@
+    ...pod.storage_pools.filter((pool) => pool.id !== pod.default_storage_pool),
+  ].filter(Boolean);
+
+  return (
+    <ContextualMenu
+      className="kvm-pool-select"
+      constrainPanelWidth
+      dropdownClassName="kvm-pool-select__dropdown"
+      dropdownContent={
+        <>
+          <div className="kvm-pool-select__grid p-contextual-menu__non-interactive">
+            <div></div>
+            <div>Storage</div>
+            <div className="u-align--right">Type</div>
+            <ul className="p-inline-list u-default-text u-no-margin--bottom">
+              <li className="p-inline-list__item">
+                <i className="p-icon--assigned is-inline"></i>
+                Allocated
+              </li>
+              <li className="p-inline-list__item">
+                <i className="p-icon--requested is-inline"></i>
+                Requested
+              </li>
+              <li className="p-inline-list__item">
+                <i className="p-icon--unassigned is-inline"></i>
+                Free
+              </li>
+            </ul>
+          </div>
+          {sortedPools.map((pool) => {
+            const isSelected = pool.name === disk.location;
+            const isDefaultPool = pool.id === pod.default_storage_pool;
+
+            // Normalise disk requests and pool storage values to GB.
+            const available = formatBytes(pool.total - pool.used, "B", {
+              convertTo: "GB",
+            }).value;
+            const allocated = formatBytes(pool.used, "B", { convertTo: "GB" })
+              .value;
+            const total = formatBytes(pool.total, "B", {
+              convertTo: "GB",
+            }).value;
+            const requested = requests[pool.name] || 0;
+            const pendingRequest = isSelected ? 0 : disk.size;
+            const free = available - requested - pendingRequest;
+
+            return (
+              <button
+                className="kvm-pool-select__pool p-button--base"
+                disabled={free < 0}
+                key={`${disk.id}-${pool.id}`}
+                onClick={() => selectPool(pool.name)}
+                type="button"
+              >
+                <div className="kvm-pool-select__grid">
+                  <div>{isSelected && <i className="p-icon--tick"></i>}</div>
+                  <div>
+                    <strong>
+                      {isDefaultPool ? `${pool.name} (default)` : pool.name}
+                    </strong>
+                    <br />
+                    <span className="u-text--light">{pool.path}</span>
+                  </div>
+                  <div className="u-align--right">
+                    {pool.type}
+                    <br />
+                    {`${total}GB`}
+                  </div>
+                  <Meter
+                    className="u-no-margin--bottom"
+                    data={[
+                      {
+                        color: "#0066CC",
```

Comment:
Might be nicer to represent these colours as variables for "allocated", "requested" etc.

Reply:
I'll add this to a colour consts file somewhere, because I'm using the same (Vanilla) colours in a couple of places